torch_dftd/functions/dftd3.py

```diff
@@ -283,21 +283,30 @@ def edisp(
             shift_abc = None if shift_abc is None else torch.cat([shift_abc, -shift_abc], dim=0)
         with torch.no_grad():
             # triplet_node_index, triplet_edge_index = calc_triplets_cycle(edge_index_abc, n_atoms, shift=shift_abc)
-            triplet_node_index, multiplicity, triplet_shift, batch_triplets = calc_triplets(
-                edge_index_abc, shift=shift_abc, dtype=pos.dtype, batch_edge=batch_edge_abc
+            # Type hinting
+            triplet_node_index: Tensor
+            multiplicity: Tensor
+            edge_jk: Tensor
+            batch_triplets: Optional[Tensor]
+            triplet_node_index, multiplicity, edge_jk, batch_triplets = calc_triplets(
+                edge_index_abc,
+                shift=shift_abc,
+                dtype=pos.dtype,
+                batch_edge=batch_edge_abc,
```

Comment:
Since this function has special cupy backend and does not have backprop, we cannot use `shift` and its derived values into this function.
So I modified its interface. Instead of returning `triplet_shift` directly, it returns indices `j` and `k` as `edge_jk`. `triplet_shift` can be calculated using `edge_jk`.